messybrainz/db/recording.py

```diff
@@ -148,3 +159,74 @@ def create_recording_clusters():
                 clusters_modified += 1
 
     return clusters_modified, clusters_add_to_redirect
+
+
+def cluster_new_recording(recording_data):
+    """ Tries to associate and cluster newly inserted recording.
+
+    Args:
+        recording_data(dict): Data present in the recording.
+
+    Returns:
+        cluster_id(UUID): Cluster ID for the new recording.
+    """
+
+    try:
+        _send_recording_to_queue(recording_data)
+    except (InternalServerError, ServiceUnavailable) as e:
+        raise
+    except Exception as e:
+        print(e)
+
+
+def _send_recording_to_queue(recording_data):
+    # check if rabbitmq connection exists or not
+    # and if not then try to connect
+    try:
+        rabbitmq_connection.init_rabbitmq_connection(current_app)
+    except ConnectionError as e:
+        current_app.logger.error('Cannot connect to RabbitMQ: %s' % str(e))
+        raise ServiceUnavailable('Cannot submit recording to queue, please try again later.')
```

Comment:
Same thing here, we should create our own Exception classes and then handle them in the webserver, raising ServiceUnavailable there.